app/models/user_group.rb

```diff
@@ -19,8 +19,12 @@ class UserGroup < ApplicationRecord
   has_many :direct_child_groups, class_name: "UserGroup", inverse_of: :parent_group, foreign_key: "parent_group_id"
   has_many :roles, foreign_key: "group_id", class_name: "UserRole"
   has_many :active_roles, -> { active }, foreign_key: "group_id", class_name: "UserRole"
+  has_many :roles_of_direct_child_groups, through: :direct_child_groups, source: :roles
+  has_many :active_roles_of_direct_child_groups, -> { active }, through: :direct_child_groups, source: :roles
```

Comment:
Maybe something like `direct_child_roles` and `active_direct_child_roles`? Not 100% confident about my suggestions, but your current naming scheme is very long as well...

Reply:
Okay, accepted the name suggestions.